FieldBuilder no longer works  with Field<Type>().Name("foo")

## Description

There is no longer an overload for Field which takes 0 arguments. We used this version of the FieldBuilder extensively unfortunately. Was this removal intentional?

```
Field<Type>().Name("foo").Resolve(...
``` 

It seems it was removed in f1f8348c0f8ae45b71f06688dfd4dde2338d5f05

### Expected result

Either a gradual deprecation or continued functionalty

### Actual result

error CS1501: No overload for method 'Field' takes 0 arguments 



### Environment

7.0.0

--- a/src/GraphQL.ApiTests/net6/GraphQL.approved.txt
+++ b/src/GraphQL.ApiTests/net6/GraphQL.approved.txt
@@ -1817,6 +1817,7 @@ namespace GraphQL.Types
             "a future release. For now you can continue to use this API but we do not encoura" +
             "ge this.")]
         public GraphQL.Types.FieldType Field(System.Type type, string name, string? description = null, GraphQL.Types.QueryArguments? arguments = null, System.Func<GraphQL.IResolveFieldContext<TSourceType>, object?>? resolve = null, string? deprecationReason = null) { }
+        [System.Obsolete("Please call Field<TGraphType>(string name) instead.")]
         public virtual GraphQL.Builders.FieldBuilder<TSourceType, object> Field<TGraphType>()
             where TGraphType : GraphQL.Types.IGraphType { }
         public virtual GraphQL.Builders.FieldBuilder<TSourceType, object> Field<TGraphType>(string name)
@@ -1830,7 +1831,10 @@ namespace GraphQL.Types
             "ge this.")]
         public GraphQL.Types.FieldType Field<TGraphType>(string name, string? description = null, GraphQL.Types.QueryArguments? arguments = null, System.Func<GraphQL.IResolveFieldContext<TSourceType>, object?>? resolve = null, string? deprecationReason = null)
             where TGraphType : GraphQL.Types.IGraphType { }
-        public virtual GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> Field<TGraphType, TReturnType>(string name = "default")
+        [System.Obsolete("Please call Field<TGraphType, TReturnType>(string name) instead.")]
+        public virtual GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> Field<TGraphType, TReturnType>()
+            where TGraphType : GraphQL.Types.IGraphType { }
+        public virtual GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> Field<TGraphType, TReturnType>(string name)
             where TGraphType : GraphQL.Types.IGraphType { }
         [System.Obsolete("Please use one of the Field() methods returning FieldBuilder and the methods defi" +
             "ned on it or just use AddField() method directly. This method may be removed in " +

--- a/src/GraphQL.ApiTests/net6/GraphQL.approved.txt
+++ b/src/GraphQL.ApiTests/net6/GraphQL.approved.txt
@@ -1817,6 +1817,8 @@ namespace GraphQL.Types
             "a future release. For now you can continue to use this API but we do not encoura" +
             "ge this.")]
         public GraphQL.Types.FieldType Field(System.Type type, string name, string? description = null, GraphQL.Types.QueryArguments? arguments = null, System.Func<GraphQL.IResolveFieldContext<TSourceType>, object?>? resolve = null, string? deprecationReason = null) { }
+        public virtual GraphQL.Builders.FieldBuilder<TSourceType, object> Field<TGraphType>()
+            where TGraphType : GraphQL.Types.IGraphType { }
         public virtual GraphQL.Builders.FieldBuilder<TSourceType, object> Field<TGraphType>(string name)
             where TGraphType : GraphQL.Types.IGraphType { }
         public virtual GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> Field<TReturnType>(string name, bool nullable = false) { }

--- a/src/GraphQL.ApiTests/netstandard20+netstandard21/GraphQL.approved.txt
+++ b/src/GraphQL.ApiTests/netstandard20+netstandard21/GraphQL.approved.txt
@@ -1817,6 +1817,7 @@ namespace GraphQL.Types
             "a future release. For now you can continue to use this API but we do not encoura" +
             "ge this.")]
         public GraphQL.Types.FieldType Field(System.Type type, string name, string? description = null, GraphQL.Types.QueryArguments? arguments = null, System.Func<GraphQL.IResolveFieldContext<TSourceType>, object?>? resolve = null, string? deprecationReason = null) { }
+        [System.Obsolete("Please call Field<TGraphType>(string name) instead.")]
         public virtual GraphQL.Builders.FieldBuilder<TSourceType, object> Field<TGraphType>()
             where TGraphType : GraphQL.Types.IGraphType { }
         public virtual GraphQL.Builders.FieldBuilder<TSourceType, object> Field<TGraphType>(string name)
@@ -1830,7 +1831,10 @@ namespace GraphQL.Types
             "ge this.")]
         public GraphQL.Types.FieldType Field<TGraphType>(string name, string? description = null, GraphQL.Types.QueryArguments? arguments = null, System.Func<GraphQL.IResolveFieldContext<TSourceType>, object?>? resolve = null, string? deprecationReason = null)
             where TGraphType : GraphQL.Types.IGraphType { }
-        public virtual GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> Field<TGraphType, TReturnType>(string name = "default")
+        [System.Obsolete("Please call Field<TGraphType, TReturnType>(string name) instead.")]
+        public virtual GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> Field<TGraphType, TReturnType>()
+            where TGraphType : GraphQL.Types.IGraphType { }
+        public virtual GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> Field<TGraphType, TReturnType>(string name)
             where TGraphType : GraphQL.Types.IGraphType { }
         [System.Obsolete("Please use one of the Field() methods returning FieldBuilder and the methods defi" +
             "ned on it or just use AddField() method directly. This method may be removed in " +

--- a/src/GraphQL.ApiTests/netstandard20+netstandard21/GraphQL.approved.txt
+++ b/src/GraphQL.ApiTests/netstandard20+netstandard21/GraphQL.approved.txt
@@ -1817,6 +1817,8 @@ namespace GraphQL.Types
             "a future release. For now you can continue to use this API but we do not encoura" +
             "ge this.")]
         public GraphQL.Types.FieldType Field(System.Type type, string name, string? description = null, GraphQL.Types.QueryArguments? arguments = null, System.Func<GraphQL.IResolveFieldContext<TSourceType>, object?>? resolve = null, string? deprecationReason = null) { }
+        public virtual GraphQL.Builders.FieldBuilder<TSourceType, object> Field<TGraphType>()
+            where TGraphType : GraphQL.Types.IGraphType { }
         public virtual GraphQL.Builders.FieldBuilder<TSourceType, object> Field<TGraphType>(string name)
             where TGraphType : GraphQL.Types.IGraphType { }
         public virtual GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> Field<TReturnType>(string name, bool nullable = false) { }

--- a/src/GraphQL.DataLoader.Tests/Types/OrderType.cs
+++ b/src/GraphQL.DataLoader.Tests/Types/OrderType.cs
@@ -13,8 +13,7 @@ public class OrderType : ObjectGraphType<Order>
         Field(x => x.OrderId);
         Field(x => x.OrderedOn);
 
-        Field<UserType, User>()
-            .Name("User")
+        Field<UserType, User>("User")
             .ResolveAsync(ctx =>
             {
                 var loader = accessor.Context.GetOrAddBatchLoader<int, User>("GetUsersById",
@@ -23,8 +22,7 @@ public class OrderType : ObjectGraphType<Order>
                 return loader.LoadAsync(ctx.Source.UserId);
             });
 
-        Field<ListGraphType<OrderItemType>, IEnumerable<OrderItem>>()
-            .Name("Items")
+        Field<ListGraphType<OrderItemType>, IEnumerable<OrderItem>>("Items")
             .ResolveAsync(ctx =>
             {
                 var loader = accessor.Context.GetOrAddCollectionBatchLoader<int, OrderItem>("GetOrderItemsById",

--- a/src/GraphQL.DataLoader.Tests/Types/QueryType.cs
+++ b/src/GraphQL.DataLoader.Tests/Types/QueryType.cs
@@ -10,8 +10,7 @@ public class QueryType : ObjectGraphType
     {
         Name = "Query";
 
-        Field<ListGraphType<UserType>, IEnumerable<User>>()
-            .Name("Users")
+        Field<ListGraphType<UserType>, IEnumerable<User>>("Users")
             .Description("Get all Users")
             .ResolveAsync(ctx =>
             {
@@ -21,8 +20,7 @@ public class QueryType : ObjectGraphType
                 return loader.LoadAsync();
             });
 
-        Field<ListGraphType<UserType>, IEnumerable<User>>()
-            .Name("UsersWithDelay")
+        Field<ListGraphType<UserType>, IEnumerable<User>>("UsersWithDelay")
             .Description("Get all Users")
             .ResolveAsync(async ctx =>
             {

--- a/src/GraphQL.DataLoader.Tests/Types/UserType.cs
+++ b/src/GraphQL.DataLoader.Tests/Types/UserType.cs
@@ -15,8 +15,7 @@ public class UserType : ObjectGraphType<User>
         Field(x => x.LastName);
         Field(x => x.Email);
 
-        Field<ListGraphType<OrderType>, IEnumerable<Order>>()
-            .Name("Orders")
+        Field<ListGraphType<OrderType>, IEnumerable<Order>>("Orders")
             .ResolveAsync(ctx =>
             {
                 var ordersLoader = accessor.Context.GetOrAddCollectionBatchLoader<int, Order>("GetOrdersByUserId",
@@ -25,8 +24,7 @@ public class UserType : ObjectGraphType<User>
                 return ordersLoader.LoadAsync(ctx.Source.UserId);
             });
 
-        Field<ListGraphType<OrderItemType>, IEnumerable<OrderItem>>()
-            .Name("OrderedItems")
+        Field<ListGraphType<OrderItemType>, IEnumerable<OrderItem>>("OrderedItems")
             .ResolveAsync(ctx =>
             {
                 //obtain a reference to the GetOrdersByUserId batch loader

--- a/src/GraphQL.Tests/Bugs/Bug2279.cs
+++ b/src/GraphQL.Tests/Bugs/Bug2279.cs
@@ -39,11 +39,11 @@ public class Bug2279GraphType<T> : ObjectGraphType<T>
     {
         if (typeof(T) == typeof(int))
         {
-            Field<IntGraphType, T>().Name("value").Resolve(x => x.Source);
+            Field<IntGraphType, T>("value").Resolve(x => x.Source);
         }
         else
         {
-            Field<StringGraphType, T>().Name("value").Resolve(x => x.Source);
+            Field<StringGraphType, T>("value").Resolve(x => x.Source);
         }
     }
 }

--- a/src/GraphQL/Types/Composite/ComplexGraphType.cs
+++ b/src/GraphQL/Types/Composite/ComplexGraphType.cs
@@ -402,7 +402,7 @@ namespace GraphQL.Types
         /// <typeparam name="TGraphType">The .NET type of the graph type of this field.</typeparam>
         /// <typeparam name="TReturnType">The return type of the field resolver.</typeparam>
         /// <param name="name">The name of the field.</param>
-        public virtual FieldBuilder<TSourceType, TReturnType> Field<TGraphType, TReturnType>(string name = "default")
+        public virtual FieldBuilder<TSourceType, TReturnType> Field<TGraphType, TReturnType>(string name)
             where TGraphType : IGraphType
         {
             var builder = CreateBuilder<TReturnType>(typeof(TGraphType)).Name(name);
@@ -410,13 +410,17 @@ namespace GraphQL.Types
             return builder;
         }
 
-        /// <summary>
-        /// Adds a new field to the complex graph type and returns a builder for this newly added field.
-        /// </summary>
-        /// <typeparam name="TGraphType">The .NET type of the graph type of this field.</typeparam>
+        /// <inheritdoc cref="Field{TGraphType, TReturnType}(string)"/>
+        [Obsolete("Please call Field<TGraphType, TReturnType>(string name) instead.")]
+        public virtual FieldBuilder<TSourceType, TReturnType> Field<TGraphType, TReturnType>()
+            where TGraphType : IGraphType
+            => Field<TGraphType, TReturnType>("default");
+
+        /// <inheritdoc cref="Field{TGraphType}(string)"/>
+        [Obsolete("Please call Field<TGraphType>(string name) instead.")]
         public virtual FieldBuilder<TSourceType, object> Field<TGraphType>()
             where TGraphType : IGraphType
-            => Field<TGraphType, object>();
+            => Field<TGraphType, object>("default");
 
         /// <summary>
         /// Adds a new field to the complex graph type and returns a builder for this newly added field.

--- a/src/GraphQL/Types/Composite/ComplexGraphType.cs
+++ b/src/GraphQL/Types/Composite/ComplexGraphType.cs
@@ -414,6 +414,14 @@ namespace GraphQL.Types
         /// Adds a new field to the complex graph type and returns a builder for this newly added field.
         /// </summary>
         /// <typeparam name="TGraphType">The .NET type of the graph type of this field.</typeparam>
+        public virtual FieldBuilder<TSourceType, object> Field<TGraphType>()
+            where TGraphType : IGraphType
+            => Field<TGraphType, object>();
+
+        /// <summary>
+        /// Adds a new field to the complex graph type and returns a builder for this newly added field.
+        /// </summary>
+        /// <typeparam name="TGraphType">The .NET type of the graph type of this field.</typeparam>
         /// <param name="name">The name of the field.</param>
         public virtual FieldBuilder<TSourceType, object> Field<TGraphType>(string name)
             where TGraphType : IGraphType

